src/main/java/bio/terra/service/snapshot/flight/export/SnapshotExportFlight.java

```diff
@@ -27,50 +31,86 @@ public SnapshotExportFlight(FlightMap inputParameters, Object applicationContext
     GcsPdao gcsPdao = appContext.getBean(GcsPdao.class);
     FireStoreDao fireStoreDao = appContext.getBean(FireStoreDao.class);
     ApplicationConfiguration appConfig = appContext.getBean(ApplicationConfiguration.class);
+    AzureBlobStorePdao azureBlobStorePdao = appContext.getBean(AzureBlobStorePdao.class);
+    ProfileService profileService = appContext.getBean(ProfileService.class);
     ObjectMapper objectMapper = appConfig.objectMapper();
 
     AuthenticatedUserRequest userReq =
         inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
     UUID snapshotId =
         UUID.fromString(inputParameters.get(JobMapKeys.SNAPSHOT_ID.getKeyName(), String.class));
 
+    var platform =
+        CloudPlatformWrapper.of(
+            inputParameters.get(JobMapKeys.CLOUD_PLATFORM.getKeyName(), String.class));
+
     boolean validatePrimaryKeyUniqueness =
         Objects.requireNonNullElse(
             inputParameters.get(ExportMapKeys.EXPORT_VALIDATE_PK_UNIQUENESS, Boolean.class), true);
 
     if (validatePrimaryKeyUniqueness) {
-      addStep(new SnapshotExportValidatePrimaryKeysStep(snapshotService, snapshotId));
+      if (platform.isGcp()) {
+        addStep(new SnapshotExportValidatePrimaryKeysStep(snapshotService, snapshotId));
+      } else {
+        throw new FeatureNotImplementedException(
```

Comment:
Can we check if the request is allowed to include `validatePrimaryKeyUniqueness` in the request validator to fail sooner?